### PR TITLE
Replace bitnami postgres with postgres operator

### PR DESCRIPTION
**Pull request by dm3ch**
/closes https://github.com/chainstack/graph-deployment/issues/17
/closes https://github.com/chainstack/graph-deployment/issues/8

Tech debt from this PR - https://github.com/chainstack/graph-deployment/issues/58

### CODE REVIEW

File: charts/graphprotocol-indexer-agent/templates/deployment.yaml [hunk 1]
```diff
@@ -62,7 +62,14 @@ spec:
             - name: INDEXER_AGENT_GRAPH_NODE_ADMIN_ENDPOINT
               value: {{ required "config.graphEndpoints.admin wasn't specified" .Values.config.graphEndpoints.admin | quote }}
             - name: INDEXER_AGENT_GRAPH_NODE_STATUS_ENDPOINT
-              value: {{ required "config.graphEndpoints.status wasn't specified" .Values.config.graphEndpoints.status | quote }}            
+              value: {{ required "config.graphEndpoints.status wasn't specified" .Values.config.graphEndpoints.status | quote }}
+            {{- if hasKey .Values.postgres.password "fromSecret" }}
+            - name: INDEXER_AGENT_POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.postgres.password.fromSecret.name }}
```
Comment (CaJIbHuK) [inline charts/graphprotocol-indexer-agent/templates/deployment.yaml:70]: Just a question: I generally ok with the proposed approach to make it possible to read password from some external secret. But can we simplify and just get rid of the `fromSecret` option? Maybe having a bit of redundancy for the sake of simplicity is not so bad idea.

File: charts/graphprotocol-node/templates/secret.yaml [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 kind: Secret
 metadata:
-  name: {{ include "graphprotocol-node.fullname" . }}-config
+  name: {{ include "graphprotocol-node.fullname" . }}
   labels:
     {{- include "graphprotocol-node.labels" . | nindent 4 }}
 data:
   config.toml: {{ include "graphprotocol-node.config" . | b64enc }}
+  {{- if not (hasKey .Values.postgres.password "fromSecret") }}
+  pgPass: {{ .Values.postgres.password.value | b64enc | quote }}
```
Comment (CaJIbHuK) [inline charts/graphprotocol-node/templates/secret.yaml:10]: The same question here

File: charts/graphprotocol-node/templates/deployment.yaml [hunk 3]
```diff
@@ -34,6 +34,13 @@ spec:
       {{- end }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: wait-for-rta
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```
Comment (CaJIbHuK) [inline charts/graphprotocol-node/templates/deployment.yaml:39]: I think we could use more lightweight image, wdyt?

File: helmfile/values/graphprotocol-indexer-agent.yaml [hunk 4]
```diff
@@ -1,7 +1,12 @@
...
-  db: graph-node
-  host: postgres-postgresql
-  user: postgres
+  db: indexer
+  host: postgres-postgres
+  user: indexer
+  password:
+    fromSecret:
+      # Get password from secret generated by postgres operator
+      name: indexer.postgres-postgres.credentials.postgresql.acid.zalan.do
```
Comment (CaJIbHuK) [inline helmfile/values/graphprotocol-indexer-agent.yaml:8]: Is this automatically generated?

Comment (CaJIbHuK) [reply helmfile/values/graphprotocol-indexer-agent.yaml:8]: If yes, then I understand the reasoning behind the previous decisions regarding the postgresql password in other charts.

**Review by CaJIbHuK — APPROVED** [review]

File: charts/graphprotocol-node/templates/deployment.yaml (hunk 3 above)
Comment (dm3ch) [reply charts/graphprotocol-node/templates/deployment.yaml:39]: We are using here same image. If we would use some lightweight image we would be pulling 2 different images wich would be less "lightweight" than using single image. :)

**Pull request merged**